homeassistant/components/recorder/__init__.py

```diff
@@ -492,10 +573,8 @@ def run(self):
                 self._shutdown()
                 return
 
-        # Start periodic purge
-        if self.auto_purge:
-            # Purge every night at 4:12am
-            track_time_change(self.hass, self.async_purge, hour=4, minute=12, second=0)
+        # Start periodic tasks
+        run_callback_threadsafe(self.hass.loop, self._async_setup_periodic_tasks)
```

Comment:
Do we need to call `result()` on the returned future?

Reply:
Actually I think we can make this even simpler.  Call `_async_setup_periodic_tasks ` from `_async_recorder_ready` and drop this `run_callback_threadsafe`

Reply:
Fixed in 11542984aa06bd43dfe49112bbf1d92e6b51297e